packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -318,14 +318,64 @@ class DateTimePicker extends React.Component {
 
   handleTimeChange(event, time) {
     this.timeValue = time;
-    const validDate = DateUtil.isValidDate(this.dateValue, this.state.dateFormat) && this.isDateTimeAcceptable(DateTimeUtils.convertDateTimeStringToMomentObject(this.dateValue, this.timeValue, this.state.dateFormat, this.props.showSeconds, this.initialTimeZone));
+    let validDate = DateUtil.isValidDate(this.dateValue, this.state.dateFormat) && this.isDateTimeAcceptable(DateTimeUtils.convertDateTimeStringToMomentObject(this.dateValue, this.timeValue, this.state.dateFormat, this.props.showSeconds, this.initialTimeZone));
     const validTime = DateTimeUtils.isValidTime(this.timeValue, this.props.showSeconds);
-    const previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+    let previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+
+    if (event.type === 'keydown') {
+      // Keyboard shortcut is pressed, split the time value into hour and minute inputs
+      const timeArray = this.timeValue.split(':');
+      if (event.key === 'n' || event.key === 'N') {
+        // `N` set Date to today
+        previousDateTime = DateUtil.createSafeDate(DateUtil.getCurrentDate(), this.initialTimeZone);
+        validDate = true;
+      } else if (event.key === '-' || event.key === '_') {
+        // `-` subtract 1 minute`
+        if (!validDate) {
+          // Date is not valid, set date to today making it valid
+          previousDateTime = DateUtil.createSafeDate(DateUtil.getCurrentDate(), this.initialTimeZone);
+          validDate = true;
```

Comment:
It is possible for consumers to pass in a date time value that is outside the range of the `minDate` and `maxDate` as illustrated in this [example](https://engineering.cerner.com/terra-framework/tests/terra-date-time-picker/date-time-picker/date-time-picker-default-date-out-of-range) with this [test](https://github.com/cerner/terra-framework/blob/terra-date-picker%404.72.0/packages/terra-date-time-picker/src/terra-dev-site/test/date-time-picker/DateTimePickerDefaultDateOutOfRange.test.jsx#L6-L8).

When this scenario occurs (i.e. [isDefaultDateAcceptable](https://github.com/cerner/terra-framework/blob/terra-date-picker%404.72.0/packages/terra-date-picker/src/DatePicker.jsx#L214) would be false), we have logic [here](https://github.com/cerner/terra-framework/blob/terra-date-picker%404.72.0/packages/terra-date-picker/src/DatePicker.jsx#L319-L323) to automatically clear the invalid initial date when the date input receives focus.

This logic appears to be broken when pressing the `N` key to populate both the date and time. The logic works on the initial input focus where both the date and time are cleared. After pressing the `N` key to set the current date and time, putting focus in the date input should no longer clear the date again for the second time. It should be cleared only once for the entire lifecycle of the component. 

Whenever we set the date and time to a valid value using any of these shortcut keys, date-time-picker needs to communicate with date-picker that the initial date is now no longer considered out of range and isDefaultDateAcceptable should no longer be false to avoid the date from being cleared a second time.

![out-of-range-date](https://user-images.githubusercontent.com/22618655/157082992-587da496-a972-4499-94fc-d4ba32cd2cde.gif)

Reply:
Added in a private prop that skips this logic in date-picker since date-time-picker also has its own version of the logic
https://github.com/cerner/terra-framework/pull/1520/commits/72af1e82737101eb86da6557fe50f9b3f1798f82#diff-2da0e35de52b21bfa57a60f6cd02cbde9957d81df2f3856f0ff48c61d68c5974R46